File: pkg01_helloworld_cpp/src/helloworld.cpp
#include <rclcpp/rclcpp.hpp>
#include <iostream>
// // 方式1 不推荐
// int main(int argc, char * argv[])
// {
//   // 初始化 ROS2
//   rclcpp::init(argc,argv);

//   // 创建节点
//   auto node = rclcpp::Node::make_shared("helloworld_node");
  
  
//   // 输出文本
//   RCLCPP_INFO(node->get_logger(),"hello world");

//   // 释放资源
//   rclcpp::shutdown();
//   return 0;
// }

// // 方式二 推荐



#include "rclcpp/rclcpp.hpp"
class MyNode: public rclcpp::Node{
public:
  MyNode():Node("node_name"){
    RCLCPP_INFO(this->get_logger(), "hello,world!(继承的方式)");
  }
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<MyNode>();

  // .....

  
  rclcpp::shutdown();
  return 0;
}

File: pkg02_helloworld_py/pkg02_helloworld_py/helloworld.py
import rclpy
from rclpy.node import Node

# # 方式一：不推荐
# def main():
#     # 初始化 ROS2 
#     rclpy.init() 
#     # 创建节点 
#     node = rclpy.create_node("helloworld_py_node") 
#     # 输出文本 
#     node.get_logger().info("杨碗振 hello world!") 
#     # 释放资源 
#     rclpy.shutdown()


# 方式二：推荐
# 自定义类：
class MyNode(Node):
    def __init__(self):
        super().__init__("node_name_py")
        self.get_logger().info("hello world! (python继承方式)")

def main():
    # 初始化 ROS2 
    rclpy.init()   
    node = MyNode()
    rclpy.shutdown()
    # 。。。。。。

if __name__ == '__main__':
    main()


